src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,180 @@
+        });
+    } else {
+      const newTodo: Todo = {
+        id: todo.id,
+        title: trimmedTitle,
+        userId: USER_ID,
+        completed: todo.completed,
+      };
+
+      const newTodos = [...todos];
+      const index = newTodos.findIndex(
+        ({ id }) => id === newTodo.id,
+      );
+
+      newTodos.splice(index, 1, newTodo);
+      updatedTodo(newTodo).then(() => {
+        setTodos(newTodos);
+        setEditTodosId(0);
+      });
+    }
+  };
+
+  const updatedComplete = () => {
+    const newTodo: Todo = {
+      id: todo.id,
+      title: todo.title,
+      userId: USER_ID,
+      completed: !todo.completed,
+    };
+
+    const newTodos = [...todos];
+    const index = newTodos.findIndex(currentTodo => (
+      currentTodo.id === newTodo.id
+    ));
+
+    newTodos.splice(index, 1, newTodo);
+
+    updatedTodo(newTodo)
+      .then(() => setTodos(newTodos));
+  };
+
+  const handlePressKey = (
+    event: React.KeyboardEvent<HTMLInputElement>,
+  ) => {
+    if (event.key === Key.Enter) {
+      updateTitle(editValue.trim());
+    } else if (event.key === Key.Escape) {
+      setEditValue(todo.title.trim());
+      setEditTodosId(0);
+    }
+  };
+
+  const handleOnBlueEditValue = () => {
+    updateTitle(editValue.trim());
+  };
+
+  const handleChangeEditValue = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setEditValue(event.target.value);
+  };
+
+  const handleDeleteValue = () => {
+    const newTodos = todos.filter(currentTodo => currentTodo.id !== todo.id);
+
+    deleteTodo(todo.id)
+      .then(() => setTodos(newTodos));
+  };
+
+  const handleDoubleClick = () => {
+    setEditTodosId(todo.id);
+    setEditValue(todo.title.trim());
+  };
+
+  return (
+    <div
+      data-cy="Todo"
+      className={classNames('todo',
+        { completed: todo.completed },
+        { editind: editTodosId === todo.id })}
+      onDoubleClick={handleDoubleClick}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          id={`toggle-view-${todo.id}`}
+          onChange={updatedComplete}
+          checked={todo.completed}
+        />
+      </label>
+
+      {editTodosId !== todo.id ? (
+        <>
+          <label data-cy="TodoTitle" className="todo__title">
+            {todo.title}
+          </label>
+
+          <button
+            type="button"
+            className="todo__remove"
+            data-cy="TodoDelete"
+            onClick={handleDeleteValue}
+          >
+            ×
+          </button>
+        </>
+      ) : (
+        <form onSubmit={(e) => e.preventDefault()}>
+          <input
+            data-cy="TodoTitleField"
+            type="text"
+            className="todo__title-field"
+            placeholder="Empty todo will be deleted"
+            value={editValue}
+            onChange={handleChangeEditValue}
+            onBlur={handleOnBlueEditValue}
```

Comment:
It's an issue (hollywar question). But as for me - Blur isn't the same as Submit when we are talking about editing. 
If user started to edit a title we have to give an option to cancel it. In this case blur is more similar to Escape, not Enter
But I can't ask you to redo this logic. Only if you think the same.